pkg/manager/manager.go

```diff
@@ -560,15 +561,52 @@ func (s *FraktiManager) Status(ctx context.Context, req *kubeapi.StatusRequest)
 func (s *FraktiManager) ListImages(ctx context.Context, req *kubeapi.ListImagesRequest) (*kubeapi.ListImagesResponse, error) {
 	glog.V(3).Infof("ListImages with request %s", req.String())
 
-	// NOTE: we only show image list of hyper runtime and assume alternative runtime images are the same
-	images, err := s.imageService.ListImages(req.GetFilter())
-	if err != nil {
-		glog.Errorf("ListImages from hyper image service failed: %v", err)
-		return nil, err
+	var (
+		lock       sync.Mutex
+		firstError error
+	)
+	catchError := func(err error) {
+		lock.Lock()
+		defer lock.Unlock()
+		if firstError == nil {
+			firstError = err
+		}
+	}
+
+	imageServiceList := []runtime.ImageService{s.imageService, s.alternativeRuntimeService}
+	serviceNameList := []string{"hyper image service", "alternative image service"}
+	imageMapList := make([]map[string]*kubeapi.Image, 2)
+
+	listImageFunc := func(i int) {
+		images, err := imageServiceList[i].ListImages(req.GetFilter())
+		if err != nil {
+			catchError(fmt.Errorf("ListImage from %s failed: %v", serviceNameList[i], err))
+			return
+		}
+		imageMapList[i] = make(map[string]*kubeapi.Image, len(images))
+		for _, image := range images {
+			imageMapList[i][image.Id] = image
+		}
+	}
+
+	workqueue.Parallelize(2, 2, listImageFunc)
+
+	if firstError != nil {
+		glog.Error(firstError)
+		return nil, firstError
+	}
+
+	// NOTE: we show intersection of image list of hyper and alternative runtime
+	intersectList := getImageListIntersection(imageMapList[0], imageMapList[1])
+
+	// if there is different in two sides, print the different if log lever is high enough
+	if glog.V(5) && len(imageMapList[0]) != len(intersectList) {
```

Comment:
We assume `getImageListDifference` may require some time to compute, so we decide only calculate it only when glog.V(5) is meet, see:  https://github.com/golang/glog/blob/master/glog.go#L992